src/Maestro/maestro-angular/src/helpers.ts

```diff
@@ -0,0 +1,43 @@
+function log(str: string) {
+  // console.log(str);
+}
+
+export function topologicalSort<TNode, TKey>(nodes: TNode[], getChildren: (node: TNode) => TNode[], getKey: (node: TNode) => TKey): TNode[] {
```

Comment:
This wouldn't be a terrible thing to add a test for. I... think it works, but it's a fairly complicated algorithm.

Reply:
ill add some.